routing/routing_tests/tools.cpp

```diff
@@ -0,0 +1,50 @@
+#include "routing/routing_tests/tools.hpp"
+
+namespace routing
+{
+void AsyncGuiThreadTestWithRoutingSession::InitRoutingSession()
+{
+  m_session = std::make_unique<routing::RoutingSession>();
+  m_session->Init(nullptr /* pointCheckCallback */);
+  m_session->SetRoutingSettings(routing::GetRoutingSettings(routing::VehicleType::Car));
+  m_session->SetOnNewTurnCallback([this]() { ++m_onNewTurnCallbackCounter; });
+}
+
+void RouteSegmentsFrom(std::vector<Segment> const & segments, std::vector<m2::PointD> const & path,
```

Comment:
Why not use 4 separate but simple and clear functions instead of this monster one?

Reply:
I would prefer creation of test routeSegments in one line.